tests/deployment/sagemaker/test_sagemaker.py

```diff
@@ -1,33 +1,419 @@
+from __future__ import absolute_import
+
+import botocore
+import pytest
+from mock import patch, MagicMock, mock_open
+from sys import version_info
+
 from botocore.exceptions import ClientError
 from botocore.stub import Stubber
 
 import boto3
+from moto import mock_ecr, mock_iam, mock_sts
 
-from bentoml.deployment.sagemaker import _parse_aws_client_exception_or_raise
+from bentoml.deployment.sagemaker import (
+    _parse_aws_client_exception_or_raise,
+    _cleanup_sagemaker_model,
+    _cleanup_sagemaker_endpoint_config,
+    get_arn_role_from_current_aws_user,
+    SageMakerDeploymentOperator,
+)
+from bentoml.proto.deployment_pb2 import Deployment
+from bentoml.proto.repository_pb2 import Bento, BentoServiceMetadata, GetBentoResponse
 from bentoml.proto.status_pb2 import Status
+from tests.deployment.sagemaker.sagemaker_moto import moto_mock_sagemaker
+
 
+@pytest.fixture(scope='function')
+def sagemaker_client():
+    return boto3.client('sagemaker', 'us-west-2')
 
-def test_sagemaker_handle_client_errors():
-    client = boto3.client('sagemaker', 'us-west-2')
-    stubber = Stubber(client)
+
+def test_sagemaker_handle_client_errors(sagemaker_client):
+    sagemaker_client = boto3.client('sagemaker', 'us-west-2')
+    stubber = Stubber(sagemaker_client)
 
     stubber.add_client_error(
         method='create_endpoint', service_error_code='ValidationException'
     )
     stubber.activate()
     result = None
-    try:
-        client.create_endpoint(EndpointName='Test', EndpointConfigName='test-config')
-    except ClientError as e:
-        result = _parse_aws_client_exception_or_raise(e)
+
+    with pytest.raises(ClientError) as error:
+        sagemaker_client.create_endpoint(
+            EndpointName='Test', EndpointConfigName='test-config'
+        )
+    result = _parse_aws_client_exception_or_raise(error.value)
 
     assert result.status_code == Status.NOT_FOUND
 
     stubber.add_client_error('describe_endpoint', 'InvalidSignatureException')
     stubber.activate()
     result = None
-    try:
-        client.describe_endpoint(EndpointName='Test')
-    except ClientError as e:
-        result = _parse_aws_client_exception_or_raise(e)
+    with pytest.raises(ClientError) as e:
+        sagemaker_client.describe_endpoint(EndpointName='Test')
+    result = _parse_aws_client_exception_or_raise(e.value)
     assert result.status_code == Status.UNAUTHENTICATED
+
+
+def test_cleanup_sagemaker_model(sagemaker_client):
+    sagemaker_client = boto3.client('sagemaker', 'us-west-2')
+    stubber = Stubber(sagemaker_client)
+    stubber.add_client_error(
+        method='delete_model', service_error_code='ValidationException'
+    )
+    stubber.activate()
+
+    error_status = _cleanup_sagemaker_model(
+        sagemaker_client, 'test_name', 'test_version'
+    )
+    assert error_status.status_code == Status.NOT_FOUND
```

Comment:
probably a follow up issue, why is ValidationException mapping to NOT_FOUND? 

I saw the code in `_parse_aws_client_exception_or_raise`:

```
    if error_code == 'ValidationException':
        logger.error(error_log_message)
        return Status.NOT_FOUND(error_response.get('Message', 'Unknown'))
```
Shouldn't it also check the actual error_message? There might be other cases when AWS returns a ValidationException right?